package.bat

```diff
@@ -1,21 +1,25 @@
+@echo on
 Rem Perform sanity checks before pip install
 pip install packaging
 Rem We use npm ci. That needs npm >= 5.7.0
 npm --version | python -c "import sys;npm_version=sys.stdin.readlines()[0].rstrip('\n');from packaging import version;supported=version.parse(npm_version) >= version.parse('5.7.0');sys.exit(1) if not supported else sys.exit(0);"
+@echo off
 if %errorlevel% neq 0 (
    echo "package.bat - ERROR: The system's npm version is not >= 5.7.0 which is required for npm ci"
    exit /b %errorlevel%
 )
-Rem uninstall packaging package since it's no longer required
-pip uninstall -y packaging
```

Comment:
Why remove this?

> Updates to package.bat to not uninstall packaging package (I was getting errors after running package.bat and then trying to build docs in windows+python+sphinx environment).

That's unrelated. The package.bat does has only one purpose and that is to package the application. If you don't remove the packaging app from the venv it will be included in the final binary and it's not needed there.

Tweaks to your setup should happen in other places and not here I guess.

Reply:
If we can't specify which packages are dev dependencies and which are not then I guess this makes sense, it just means I have to install 'packaging' again every time in order to build docs dynamically as I'm working on them.

Reply:
That's strange btw, I wonder why you need packaging to work in docs.

Okay okay. So I see. Interesting. Indeed sphinx does need packaging. Also pytest-cov has it listed as a dependency.

> If we can't specify which packages are dev dependencies 

In theory yes, and in theory pyinstaller should do it automatically. But sometimes it does not work and we end up with extra stuff in the bundled pyinstaller binary.

So not sure if this will work ... but you can try to add packaging to the excludes here:

https://github.com/rotki/rotki/blob/2851d421ca9620ae0a36316079c5d1ca3bee3d3c/rotkehlchen.spec#L91

Also any changes to package.bat (the echo stuff are fine) should be reflected in package.sh. It's the same script just as a shell script for Linux and OSX:

https://github.com/rotki/rotki/blob/2851d421ca9620ae0a36316079c5d1ca3bee3d3c/package.sh#L15